sync/src/relayer/mod.rs

```diff
@@ -558,21 +574,24 @@ impl Relayer {
             }
         }
 
-        let send_block_proposals =
-            |nc: &dyn CKBProtocolContext, peer_index: PeerIndex, txs: Vec<packed::Transaction>| {
-                let content = packed::BlockProposal::new_builder()
-                    .transactions(txs)
-                    .build();
-                let message = packed::RelayMessage::new_builder().set(content).build();
-                let status = quick_send_message_to(nc, peer_index, &message);
-                if !status.is_ok() {
-                    ckb_logger::error!(
-                        "send RelayBlockProposal to {}, status: {:?}",
-                        peer_index,
-                        status
-                    );
-                }
-            };
+        async fn send_block_proposals(
```

Comment:
extract it to a helper fn and reuse with https://github.com/nervosnetwork/ckb/blob/5c6bfca16eb5b1b933128e531589be0dead8b2ba/sync/src/relayer/get_block_proposal_process.rs#L99